tests/test_graph.py

```diff
@@ -30,18 +80,116 @@ def test_valid_recipe_order_and_bootstrap(names, bootstrap):
 def test_invalid_recipe_order_and_bootstrap(names, bootstrap):
     with pytest.raises(BuildInterruptingException) as e_info:
         get_recipe_order_and_bootstrap(ctx, names, bootstrap)
-    assert e_info.value.message == (
-        "Didn't find any valid dependency graphs. "
-        "This means that some of your requirements pull in conflicting dependencies."
+    assert "conflict" in e_info.value.message.lower()
+
+
+@pytest.mark.parametrize('names,bootstrap', valid_combinations)
+def test_valid_obvious_conflict_checker(names, bootstrap):
+    # Note: obvious_conflict_checker is stricter on input
+    # (needs fix_deplist) than get_recipe_order_and_bootstrap!
+    obvious_conflict_checker(ctx, fix_deplist(names))
+
+
+@pytest.mark.parametrize('names,bootstrap',
+                         invalid_combinations_simple  # see above for why this
+                        )                             # is a separate list
+def test_invalid_obvious_conflict_checker(names, bootstrap):
+    # Note: obvious_conflict_checker is stricter on input
+    # (needs fix_deplist) than get_recipe_order_and_bootstrap!
+    with pytest.raises(BuildInterruptingException) as e_info:
+        obvious_conflict_checker(ctx, fix_deplist(names))
+    assert "conflict" in e_info.value.message.lower()
+
+
+def test_misc_obvious_conflict_checker(monkeypatch):
+    # Check that the assert about wrong input data is hit:
+    with pytest.raises(AssertionError) as e_info:
+        obvious_conflict_checker(
+            ctx,
+            ["this_is_invalid"]
+            # (invalid because it isn't properly nested as tuple)
+        )
+
+    # Test that non-recipe dependencies work in overall:
+    obvious_conflict_checker(
+        ctx, fix_deplist(["python3", "notarecipelibrary"])
     )
 
+    # Test that a conflict with a non-recipe dependency works:
+    # This is currently not used, so we need a custom test recipe:
+    # To get that, we simply modify one!
+    with monkeypatch.context() as m:
+        register_fake_recipes_for_test(m, [
+            get_fake_recipe("recipe1", conflicts=[("fakelib")]),
+        ])
+        with pytest.raises(BuildInterruptingException) as e_info:
+            obvious_conflict_checker(ctx, fix_deplist(["recipe1", "fakelib"]))
+        assert "conflict" in e_info.value.message.lower()
+
+    # Test a case where a recipe pulls in a conditional tuple
+    # of additional dependencies. This is e.g. done for ('python3',
+    # 'python2', ...) but most recipes don't depend on this anymore,
+    # so we need to add a manual test for this case:
+    with monkeypatch.context() as m:
+        register_fake_recipes_for_test(m, [
+            get_fake_recipe("recipe1", depends=[("libffi", "Pillow")]),
+        ])
+        obvious_conflict_checker(ctx, fix_deplist(["recipe1"]))
```

Comment:
This test won't say if this feature is working or not, it currently just check that the code is not crashing, right? There're few other tests like that here and there. Can you think of if it's something we can improve?

Reply:
@AndreMiras it checks more because `obvious_conflict_checker` can raise `BuildInterruptingException`. So if the test uses that `with` construct to expect that it tests/expects a conflict result, otherwise it tests/expects a non-conflict result. So it's definitely "evaluating" the result at least in the most basic way

Reply:
Yes this is what I mean by crashing. I mean that this test just verify some code path is not raising any exception, nothing more.
So my point is instead of doing that check can you think of a way to prove maybe another point with this test? Basically which code path did you have in mind and can we prove it went through. That's my point

Reply:
Maybe it could check more at some point but I'm pretty okay with what it does now. I can't think at least of an obvious way to check it more granular, although surely there is always one way or another to improve it

Reply:
Yes I'm just challenging you, we covered quite some cases with that PR, so kudos to you!